### PR TITLE
MB-2112 TOO can see MTO Service Items that are SUBMITTED, APPROVED, or REJECTED

### PR DESCRIPTION
**Pull request by Ryan-Koch**
## Description

Makes it so the `RequestServiceItemsTable` component can handle `mtoServiceItems` of the three status types: `SUBMITTED`, `APPROVED`, and `REJECTED`. Also implements a button to allow the TOO to undo rejections by approving the item. The ability to reject a previously approved item won't be possible until after https://dp3.atlassian.net/browse/MB-1482 is played.

## Setup

To get some data into your environment:

```sh
make db_dev_e2e_populate
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-2112) for this change

## Screenshots

image

### CODE REVIEW

File: src/components/Office/RequestedServiceItemsTable/RequestedServiceItemsTable.jsx [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
 import PropTypes from 'prop-types';
 
+import { SERVICE_ITEM_STATUS } from '../../../shared/constants';
+
 import styles from './RequestedServiceItemsTable.module.scss';
 
 import ServiceItemTableHasImg from 'components/ServiceItemTableHasImg';
 
 const RequestedServiceItemsTable = ({ serviceItems, handleUpdateMTOServiceItemStatus }) => {
+  const chooseTitleText = (serviceItem) => {
```
Comment (duncan-truss) [inline src/components/Office/RequestedServiceItemsTable/RequestedServiceItemsTable.jsx:11]: I like reusing this component for all the tables.  I think we already know which status each is representing though so not sure we need to inspect the service items again.  I would just pass the status as a prop from MoveTaskOrder probably.

Comment (Ryan-Koch) [reply src/components/Office/RequestedServiceItemsTable/RequestedServiceItemsTable.jsx:11]: Ah that's a good point, I'll change this up to use a passed in prop from the page level.

File: src/components/ServiceItemTableHasImg/index.jsx [hunk 2]
```diff
@@ -52,31 +52,67 @@ const ServiceItemTableHasImg = ({ serviceItems, handleUpdateMTOServiceItemStatus
...
+              </Button>
+            </div>
+          )}{' '}
+          {status === SERVICE_ITEM_STATUS.APPROVED && (
+            <div className={styles.statusAction}>
               <span className="icon">
-                <Check />
+                <Ex />
+              </span>
+              <span
```
Comment (duncan-truss) [inline src/components/ServiceItemTableHasImg/index.jsx:87]: USWDS has an unstyled button that may be worth using over trying to add functionality to a span https://designsystem.digital.gov/components/button/

Comment (Ryan-Koch) [reply src/components/ServiceItemTableHasImg/index.jsx:87]: Oh nice! I overlooked this. Would save some of this extra effort around trying to force this to be reasonable accessible.

Comment (Ryan-Koch) [reply src/components/ServiceItemTableHasImg/index.jsx]: Went ahead and used the react-uswds component and passed in that unstyled class.

File: src/components/Office/RequestedServiceItemsTable/RequestedServiceItemsTable.jsx (hunk 1 above)
Comment (Ryan-Koch) [reply src/components/Office/RequestedServiceItemsTable/RequestedServiceItemsTable.jsx]: Done!

**Comment by duncan-truss** [comment]
@Ryan-Koch I'm not seeing the approve and reject buttons for the requested service items table anymore in the loki images.

**Comment by Ryan-Koch** [comment]
> @Ryan-Koch I'm not seeing the accept and reject buttons for the requested service items table anymore in the loki images.

I think they get cut off on the mobile views, they seem to be showing up in the larger screenshots and in storybook when I look at it.

**Comment by duncan-truss** [comment]
> > @Ryan-Koch I'm not seeing the accept and reject buttons for the requested service items table anymore in the loki images.
> 
> I think they get cut off on the mobile views, they seem to be showing up in the larger screenshots and in storybook when I look at it.

It's the `ShipmentContainer - MTO Accessorial` story and the `Tables - Service Item Tables` that need the `statusForTableType` prop added to their stories.

**Comment by Ryan-Koch** [comment]
> > > @Ryan-Koch I'm not seeing the accept and reject buttons for the requested service items table anymore in the loki images.
> > 
> > 
> > I think they get cut off on the mobile views, they seem to be showing up in the larger screenshots and in storybook when I look at it.
> 
> It's the `ShipmentContainer - MTO Accessorial` story and the `Tables - Service Item Tables` that need the `statusForTableType` prop added to their stories.

Oh I see what you're talking about. I think I got it now.

File: pkg/services/mto_service_item/mto_service_item_updater.go [hunk 3]
```diff
@@ -38,10 +38,6 @@ func (p *mtoServiceItemUpdater) UpdateMTOServiceItemStatus(mtoServiceItemID uuid
 		return nil, services.NewNotFoundError(mtoServiceItemID, "MTOServiceItemID")
 	}
 
-	if mtoServiceItem.Status != models.MTOServiceItemStatusSubmitted || (status != models.MTOServiceItemStatusApproved && status != models.MTOServiceItemStatusRejected) {
-		return nil, services.NewConflictError(mtoServiceItemID, "MTOServiceItemID")
-	}
-
```
Comment (monfresh) [inline pkg/services/mto_service_item/mto_service_item_updater.go:44]: @Ryan-Koch What was the reason for removing this? Do we no longer need state transition checks or do we have them elsewhere?

Comment (Ryan-Koch) [reply pkg/services/mto_service_item/mto_service_item_updater.go:44]: After some conversation with design it was decided that we don't need to be preventing folks from undoing an approval or rejection (we want them to be able to make a change if they misclick). For now we've decided to go the most permissive route by not having the check. In the future when we have a status for the overall move to say that it's closed out we may want to adjust a bit and add something to check at that level.

**Pull request merged**